### PR TITLE
Add check to verify copyright file exists for all platform packages.

### PR DESCRIPTION
**Pull request by smeda-delphix**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build was run locally and any changes were pushed
- [ ] Lint has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: N/A

## What is the new behavior?
Each custom platform package would need to have copyright file in the Debian package for us to generate correct license information and BOM for 3rd party packages. Add check to verify the copyright file is present in right location before building packages.

Verified running locally each package to see copyright file is being detected.
Whitelisted packages that are not build from repositories and are not 3rd party packages.

Addressed comments from https://github.com/delphix/linux-pkg/pull/108

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: default-package-config.sh [hunk 1]
```diff
@@ -300,3 +300,23 @@ function kernel_update_upstream() {
...
+
+	# This function checks for SKIP_COPYRIGHTS_CHECK flag
+	# in config.sh file of each package. If the flag is
+	# present and is set to 'true', the check will be skipped.
+	# The license information for the platform packages are
+	# generated based on Ubuntu package convention and are
+	# picked from copyright file under /debian folder. As a
+	# part of the check we look for existance of the file in
+	# each package.
+	if [[ ! -n $SKIP_COPYRIGHTS_CHECK || -n $SKIP_COPYRIGHTS_CHECK && "$SKIP_COPYRIGHTS_CHECK" != true ]]; then
```
Comment (pzakha) [inline default-package-config.sh:314]: You can just write:
```if [[ "$SKIP_COPYRIGHTS_CHECK" != "true" ]]```

File: default-package-config.sh [hunk 2]
```diff
@@ -300,3 +300,23 @@ function kernel_update_upstream() {
...
+		echo "Start copyright check"
+		file_count=$(find "$WORKDIR/repo" | grep 'debian/copyright' -c)
+
+		if [[ ! $file_count -gt 0 ]]; then
```
Comment (pzakha) [inline default-package-config.sh:318]: Would this work instead:
```if [[ ! -f "$WORKDIR/repo/debian/copyright" ]]```

Comment (smeda-delphix) [reply default-package-config.sh:318]: The path is not same for all repos, especially if the repo is from gitlab. 
Packages from gitlab have some extra folders before /debian

Comment (pzakha) [reply default-package-config.sh:318]: Ah okay, makes sense

**Pull request merged**